src/tests/unit/auto/auto_runtime_fallback_test.cpp

```diff
@@ -0,0 +1,281 @@
+    std::tie(targetDevices, targetDevicesThrow, loadNetworkNum, enableRumtimeFallback) = this->GetParam();
+
+    for (auto& deviceName : targetDevices) {
+        targetDev += deviceName;
+        targetDev += ((deviceName == targetDevices.back()) ? "" : ",");
+    }
+    plugin->SetName("AUTO");
+    config.insert({InferenceEngine::MultiDeviceConfigParams::KEY_MULTI_DEVICE_PRIORITIES, targetDev});
+    if (!enableRumtimeFallback) {
+        config.insert({{"ENABLE_RUNTIME_FALLBACK", "NO"}});
+    }
+    if (targetDevices.size() == 2) {
+        mockInferrequest = std::make_shared<mockAsyncInferRequest>(
+            inferReqInternal, mockExecutor, nullptr, targetDevicesThrow[1]);
+        ON_CALL(*mockIExeNet.get(), CreateInferRequest()).WillByDefault(Return(mockInferrequest));
+
+        mockInferrequestActual = std::make_shared<mockAsyncInferRequest>(
+            inferReqInternalActual, mockExecutorActual, nullptr, targetDevicesThrow[0]);
+        ON_CALL(*mockIExeNetActual.get(), CreateInferRequest()).WillByDefault(Return(mockInferrequestActual));
+
+        EXPECT_CALL(*core,
+                    LoadNetwork(::testing::Matcher<const InferenceEngine::CNNNetwork&>(_),
+                                ::testing::Matcher<const std::string&>(_),
+                                ::testing::Matcher<const std::map<std::string, std::string>&>(_)))
+            .Times(loadNetworkNum);
+    } else if (targetDevices.size() == 3) {
+        mockInferrequest = std::make_shared<mockAsyncInferRequest>(
+            inferReqInternal, mockExecutor, nullptr, targetDevicesThrow[2]);
+        ON_CALL(*mockIExeNet.get(), CreateInferRequest()).WillByDefault(Return(mockInferrequest));
+
+        mockInferrequestActual = std::make_shared<mockAsyncInferRequest>(
+            inferReqInternalActual, mockExecutorActual, nullptr, targetDevicesThrow[0]);
+        ON_CALL(*mockIExeNetActual.get(), CreateInferRequest()).WillByDefault(Return(mockInferrequestActual));
+
+        mockInferrequestActualBackUp = std::make_shared<mockAsyncInferRequest>(
+            inferReqInternalActualBackUp, mockExecutorActualBackUp, nullptr, targetDevicesThrow[1]);
+        ON_CALL(*mockIExeNetActualBackUp.get(), CreateInferRequest()).WillByDefault(Return(mockInferrequestActualBackUp));
+
+        EXPECT_CALL(*core,
+                    LoadNetwork(::testing::Matcher<const InferenceEngine::CNNNetwork&>(_),
+                                ::testing::Matcher<const std::string&>(_),
+                                ::testing::Matcher<const std::map<std::string, std::string>&>(_)))
+            .Times(loadNetworkNum);
+    }
+
+    std::shared_ptr<InferenceEngine::IExecutableNetworkInternal> exeNetwork;
+
+    ASSERT_NO_THROW(exeNetwork = plugin->LoadExeNetworkImpl(cnnNet, config));
+
+    std::shared_ptr<IInferRequestInternal> infer_request;
+    ASSERT_NO_THROW(infer_request = exeNetwork->CreateInferRequest());
+    ASSERT_NO_THROW(infer_request->StartAsync());
```

Comment:
when disable ENABLE_RUNTIME_FALLBACK, to me, some exceptions should be expected, while I did not see this situation been tested?
some test scenarios maybe:
1) runtime_fallback enabled
    a. helper loaded first, helper failed
    b. helper loaded first, actual failed
    c. actual loaded first, actual failed
    d. helper been disabled, actual failed
    d. cumulative situation(TBD)
2) runtime_fallback disabled
    exceptions will be seen no matter which failed

Reply:
In CTPUT situation, if inference on one device fail, AUTO remove this device from worker list and infer current input on another device.

For example, AUTO load network to GPU.1, GPU.0, VPUX and CPU. If inference on VPUX fail, AUTO remove VPUX from worker list, current input in pipeline will infer in next worker.

Reply:
Added these cases.